### PR TITLE
sea-orm-cli: init at 0.9.1

### PR DESCRIPTION
**Pull request by lucperkins**
###### Description of changes

This PR adds a package for the CLI tool for [Sea ORM](https://www.sea-ql.org/SeaORM) a popular object-relational mapper for Rust.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/sea-orm-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-OEsUNicNRfXUctpuVAb/3Raa+x6qeg9Mtx9FtHzBcjw=";
+  };
+
+  cargoSha256 = "sha256-aiUk+3Ppn+AgezNnK7s3AlmFTduH8stg3oxHbbczBE0=";
+
+  cargoBuildFlags = [ "--bin" pname ];
+
+  buildInputs = lib.optionals stdenv.isDarwin [ SystemConfiguration ];
```
Comment (lelgenio) [inline pkgs/development/tools/rust/sea-orm-cli/default.nix:16]: It looks like it's missing openssl for linux support.
```suggestion
  nativeBuildInputs = [ pkg-config ];
  buildInputs = [ openssl ]
    ++  lib.optionals stdenv.isDarwin [ SystemConfiguration ];
```

**Comment by lucperkins** [comment]
@lelgenio Good catch!

**Comment by tengkuizdihar** [comment]
Nice, someone else is working on this XD, also, what's holding up the PR? Currently it's on version `0.10.1`, would it be possible to bump the version in this PR as well?

**Review by lelgenio — APPROVED** [review]
Working great on x86_64-linux. One thing to note is that the executable named `sea` is not installed, which does not match what happens when installed via cargo, which is fine I guess.

**Review by Jumail — APPROVED** [review]

**Comment by chayleaf** [comment]
#231339 is now merged, so this should be closed

**Pull request closed without merging**